src/lib.rs

```diff
@@ -29,62 +40,69 @@ impl <const W: usize, const H: usize> Content<W, H> {
 /// This receives the `SmartLedsWriter` trait implementations along with a
 /// `MatrixType` that describes the size and the pixels mapping between the LED
 /// strip placement and the matrix's x y coordinates.
-pub struct SmartLedMatrix<T, M: MatrixType, const W: usize, const H: usize> {
+pub struct SmartLedMatrix<T, M: MatrixType<W, H>, const W: usize, const H: usize> {
     writer: T,
     content: Content<W, H>,
-    matrix_type: M,
+    matrix_type: PhantomData<M>,
```

Comment:
By using `PhantomData` you don't need to pass a the matrix type parameter to the `new` constructor. But this also means that the `MatrixType` implementation cannot store any state. IMO this shouldn't be a problem.